code/datums/elements/glass_bleeder.dm

```diff
@@ -0,0 +1,24 @@
+/// Drop glass shards when taking damage
+/datum/element/glass_bleeder
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY
+
+/datum/element/glass_bleeder/Attach(atom/movable/target)
+	. = ..()
+
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_LIVING_ADJUST_BRUTE_DAMAGE, PROC_REF(on_adjust_brute_damage))
```

Comment:
You should probably use the apply_damage signal instead
it's 1. a more natural signal to use and 2. doesn't trigger on healing, only damage